tests/workflows/test_consistency.py

```diff
@@ -0,0 +1,235 @@
+    and each of the settings coming from settings_list. Standard TSNR is calculated inside thefmriprep workflows.
+    We create a zip file with all the features, tsnr and spec file, per participant.
+    """
+
+    dataset_file = dataset.download(tmp_path)
+
+    spec = make_spec(dataset_files=[dataset_file], pcc_mask=pcc_mask, test_settings=settings_list)
+    config.nipype.omp_nthreads = cpu_count()
+    save_spec(spec, workdir=tmp_path)
+
+    # Run built-in halfpipe
+    workflow = init_workflow(tmp_path)
+    graphs = init_execgraph(tmp_path, workflow)
+    parser = build_parser()
+    opts = parser.parse_args(args=list())
+    opts.graphs = graphs
+    opts.debug = True
+    opts.nipype_run_plugin = "MultiProc"  # "Simple", "MultiProc"
+    opts.workdir = tmp_path  # necessary when you choose MultiProc
+
+    run_stage_run(opts)
+    # OR: Run another halfpipe version (not implemented)
+
+    # subprocess.call(["docker", "run", "--volume", f"{tmp_path}:{tmp_path}", "--rm", "halfpipe/halfpipe:1.2.2 "])
+
+    index = BIDSIndex()
+    index.put(tmp_path / "derivatives")
+    spec_file = tmp_path / "spec.json"
+
+    for sub in dataset.subject_ids:
+        paths_to_zip = []
+        for test_setting in settings_list:
+            name = test_setting.name
+
+            # Search for files we want to save in each setting
+            timeseries = index.get(sub=sub, feature=f"{name}CorrMatrix", suffix="timeseries", task="rest", extension=".tsv")
+            corr = index.get(sub=sub, feature=f"{name}CorrMatrix", desc="correlation", suffix="matrix")
+            dual_reg = index.get(sub=sub, feature=f"{name}DualReg", suffix="statmap", stat="z", component="8")
+            falff = index.get(sub=sub, feature=f"{name}FALFF", suffix="falff", extension=".nii.gz")
+            alff = index.get(sub=sub, feature=f"{name}FALFF", suffix="alff", extension=".nii.gz")
+            reho = index.get(sub=sub, feature=f"{name}ReHo", suffix="reho", extension=".nii.gz")
+            pcc = index.get(sub=sub, feature=f"{name}SeedCorr", suffix="statmap", stat="z")
+
+            # Check that halfpipe processing generated those files
+            for feature_name, feature_path in [
+                ("timeseries", timeseries),
+                ("correlation matrix", corr),
+                ("dual regression", dual_reg),
+                ("falff", falff),
+                ("alff", alff),
+                ("reho", reho),
+                ("pcc", pcc),
+            ]:
+                assert feature_path is not None and len(feature_path) == 1, f"Missing path for {name} {feature_name}"
+
+            paths_to_zip.extend(
+                [
+                    list(timeseries or [])[0],
+                    list(corr or [])[0],
+                    list(dual_reg or [])[0],
+                    list(falff or [])[0],
+                    list(alff or [])[0],
+                    list(reho or [])[0],
+                    list(pcc or [])[0],
+                ]
+            )
```

Comment:
Can you try refactoring this to use one loop that does the `index.get`, the assertion and the `extend` all in one? I think this would make the control flow clearer and also make it easier to read what the differences are between the different function calls.